File: src/hackerrank/FlippingTheMatrix.java
package hackerrank;

import java.util.List;

public class FlippingTheMatrix {
    public static int flippingMatrix(List<List<Integer>> matrix) {
        // Write your code here
        int sum = 0;
        int lastIndex = matrix.size() - 1;
        for (int i = 0; i < matrix.size() / 2; i++) {
            for (int j = 0; j < matrix.size() / 2; j++) {
                int upper = Math.max(matrix.get(i).get(j), matrix.get(i).get(lastIndex - j));
                int lower = Math.max(matrix.get(lastIndex - i).get(j), matrix.get(lastIndex - i).get(lastIndex - j));
                sum += Math.max(upper, lower);
            }
        }
        return sum;
    }
}

File: src/hackerrank/FlippingTheMatrix.kt
package hackerrank

class FlippingTheMatrix {
    fun flippingMatrix(matrix: Array<Array<Int>>): Int {
        var sum = 0
        for (i in 0 until matrix.size / 2) {
            for (j in 0 until matrix.size / 2) {
                val a = matrix[i][j]
                val b = matrix[i][matrix.size - 1 - j]
                val c = matrix[matrix.size - 1 - i][j]
                val d = matrix[matrix.size - 1 - i][matrix.size - 1 - j]
                sum += listOf(a, b, c, d).max()!!
            }
        }

        return sum
    }
}